tests/roles/ovn_adoption/tasks/ospdo_ovn_adoption.yaml

```diff
@@ -0,0 +1,37 @@
+- name: get OSPdO OVN ovsdb-server IP
+  delegate_to: "{{ adoption_ospdo_host }}"
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    oc get cm -o yaml|grep ovn_dbs_node_ips|tr -d ' '|sed 's/.*ovn_dbs_node_ips:\\n-//g'|sed 's/\\n.*//g'
+  register: source_ovndb_ip
+  when: get_source_ovndb_ip| default(false) | bool
+
+- name: Fetch OVN databases, cp into helper pod
+  block:
```

Comment:
you guard the previous task with "get_source_ovndb_ip" on line 7

so if that is set to false, then this following block will fail becuase source_ovndb_ip will be undefined.

so maybe a conditional on this block?

when: source_ovndb_ip is defined?
or even simpler put the previous task inside the block and just have the same condition as above but for the block:

- name: Fetch OVN databases, cp into helper pod
   when: get_source_ovndb_ip| default(false) | bool
   block:

Reply:
the   register: source_ovndb_ip task is used in the next task:
- name: set OVN copy shell vars , and that creates a variable that is used in the 
- name: start an adoption helper pod task that later is used lastly by my the - name: Stop Fetch OVN databases, cp into helper pod
block . So I to not interrupt the original flow it seems to me the best approach of safeguarding this is by: adding an and source_ovndb_ip is defined in the block's when list.